File: android/app/src/main/klotin/com/example/firebolt/Payjoin.kt
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import com.yourpackagename.payjoin.PayJoinService

class MainActivity : AppCompatActivity() {
    private val payJoinService = PayJoinService()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.payJoinButton).setOnClickListener {
            GlobalScope.launch {
                val transactionData = Any() // Prepare transaction data
                val response = payJoinService.initiatePayJoinTransaction(transactionData)
                // Handle response
            }
        }
    }
}

File: ios/app/payjoin.swift
import UIKit
import PayJoin

class ViewController: UIViewController {
    let payJoinService = PayJoinService()

    override func viewDidLoad() {
        super.viewDidLoad()

        let payJoinButton = UIButton(type: .system)
        payJoinButton.setTitle("Initiate PayJoin", for: .normal)
        payJoinButton.addTarget(self, action: #selector(initiatePayJoin), for: .touchUpInside)
        view.addSubview(payJoinButton)
    }

    @objc func initiatePayJoin() {
        DispatchQueue.global().async {
            let transactionData = Any() // Prepare transaction data
            let response = self.payJoinService.initiatePayJoinTransaction(transactionData: transactionData)
            // Handle response
        }
    }
}